spec/03-plugins/06-statsd/01-log_spec.lua

```diff
@@ -655,6 +665,27 @@ for _, strategy in helpers.each_strategy() do
     end)
 
     describe("metrics", function()
+      -- the purpose of this test case is to test the batch queue
+      -- finishing processing its batch in one time (no retries)
+      it("no more metrics than expected", function()
+        local metrics_count = DEFAULT_METRICS_COUNT + shdict_count * 2
+
+        local thread = helpers.udp_server(UDP_PORT, metrics_count + 1, 5)
+        local response = assert(proxy_client:send {
+          method  = "GET",
+          path    = "/request?apikey=kong",
+          headers = {
+            host  = "logging31.com"
+          }
+        })
+        assert.res_status(200, response)
+
+        local ok, metrics, err = thread:join()
+        assert(ok, metrics)
+        assert(#metrics == metrics_count, err)
+        ngx.sleep(DEFAULT_SHDICT_METRICS_SEND_THRESHOLD + 1)
```

Comment:
This sleep is too long. It sleeps for 60 seconds which makes the tests run extremely slowly. Can we improve the method?

Reply:
The reason why it has to sleep is because shdict metrics won't be sent again until the pulling interval exceeds the SHDICT_METRICS_SEND_THRESHOLD. I‘m considering separating it to another independent descirbe unit.